부산대_8조_위키키_7주차

kimdonwoo opened the pull request:
## ⭐Key Changes
1. 페이지 링크 거는 API에서 redis로 성능 개선
2. 글 신고 기능 구현
3. 문서 기여 목록 페이지 구현등등

<br>

## 👪멘토님께
> 7주차에는 팀원 모두 졸업과제 마무리한다고 코드는 많이 못짰습니다!!

kimdonwoo committed:
Feat : List<Post>를 입력받아 목차를 생성하는 IndexUtils 구현

- Tree 형식으로 구현된 List<Post>를 입력받아 DFS로 목차를 생성하는 IndexUtils 구현하였습니다.

kimdonwoo committed:
Feat : 페이지ID로 페이지 조회 구현

- 페이지 ID를 입력받아 특정 페이지들의 Post들을 조회하는 API 구현하였습니다.

kimdonwoo committed:
Feat : Post 생성시 History 생성되도록 구현

- 페이지에 Post 생성시 History도 생성되도록 구현하였습니다.
- Post 생성시 Page의 updated_at 수정되도록 구현하였습니다.

kimdonwoo committed:
Feat : Post 내용 수정 API 구현

- 기존 Post에 있는 내용들과 동일하면 Exception을 던져주도록 구현하였습니다.
- Post의 내용이 변경되면 History 테이블에 저장되도록 구현하였습니다.

kimdonwoo committed:
Style : 페이지 조회시 목차 구조 변경

- 페이지 조회 시 기존 1-1-1 방식에서 1.1.1로 수정

kimdonwoo committed:
Feat : 최근 바뀐 페이지 조회 API 구현

- 최근 생성/변경이 있는 10개의 페이지만 조회하도록 구현하였습니다.
- 현재 group쪽은 연결이 안되어 있어서 where문에 groupId는 추후에 추가할 예정입니다.

kimdonwoo committed:
Feat : 페이지 생성에서 동일한 title에 대한 Exception 추가

- 이미 존재하는 Title의 페이지 생성시 예외 처리
- 추후에 where 문에 groupId 추가할 예정

kimdonwoo committed:
Feat : 페이지 삭제 API 구현

- 해당 페이지에 post가 존재하는지 확인하고 존재하면 ApplicationException을 던지도록 구현하였습니다.
- post가 존재안하는 페이지만 삭제 가능하도록 구현하였습니다.

kimdonwoo committed:
Feat : 페이지 제목으로 페이지 조회 API 구현

- groupId랑 title을 가지고 특정 그룹에 존재하는 title의 페이지를 조회할 수 있는 API 구현하였습니다.

kimdonwoo committed:
Feat : 단어에 링크를 걸기 위한 PagdId 조회 API 구현

- 글 작성시 단어에 링크를 걸기 위한 PageId 조회하는 API 구현하였습니다.

kimdonwoo committed:
Feat : Post 삭제 기능 구현

- Post 삭제 시에 해당 Post의 Id를 가진 history 객체들도 모두 삭제시켜주었습니다
- Post들이 한 페이지내에서 트리 구조로 이루어져있어서 child Post를 가지고 있지 않는 Post들만 삭제가능하도록 예외처리하였습니다.
- Post들 사이에 순서가 존재하므로 중간 순서의 Post가 삭제되었을 때, 뒷 순서의 Post들의 orders를 -1 처리해주었습니다.

Jun2-Lee committed:
Fix : pull 이후 오류 수정 및 MemberId Long으로 변경
- pull 이후 error 패키지 안 오류 수정
- MemberId Long으로 변경
- jwt 토큰 안의 getName을 email이 아닌 id로 수정

Jun2-Lee committed:
Merge pull request #18 from Step3-kakao-tech-campus/junhee

배포를 위한 에러 해결

kimdonwoo committed:
Merge branch '6-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into donwoo

 Conflicts:
	wekiki/src/main/java/com/kakao/techcampus/wekiki/history/History.java

kimdonwoo committed:
Feat : 페이지 목차 조회 기능 구현

- 프론트팀원의 요청으로 페이지 목차 조회 API 추가하였습니다.

kimdonwoo committed:
Refactor : 페이지 이름 pageName으로 통일

- pageName이랑 title을 혼동해서 쓰는 경우가 있어서 모두 pageName으로 통일하였습니다.

kimdonwoo committed:
Refactor : 페이지 제목으로 페이지 조회 API에서 response 수정

- 프론트 팀원의 요청으로 페이지 제목으로 페이지 조회 api에서 response에 pageId도 추가해주었습니다.

kimdonwoo committed:
Feat : 글 히스토리 조회 구현

- 특정 post의 히스토리 조회 기능을 구현하였습니다.
- 페이지네이션을 적용하여 5개씩 response에 담았습니다.

kimdonwoo committed:
Feat : 페이지 키워드 검색 기능에서 페이지 content 추가

- 페이지 키워드에 검색에서 페이지의 가장 위에 있는 post의 내용을 responseDTO의 content에 추가하였습니다.
- post가 하나도 존재하지 않는 페이지는 빈 문자열을 반환해줍니다.

kimdonwoo committed:
Feat : 페이지/글 관련 url 매핑 수정

- PageRestController에서 RequestMapping에 "/group/{groupid}/page"로 수정하였습니다.
- PostRestController에서 RequestMapping에 "/group/{groupid}/post"로 수정하였습니다.

INUK-ai committed:
Feat: 초대 링크 판별 기능 구현

- 초대 링크 식별 후 GroupId 반환
- 휘발시킬지 말지 고민 중

kimdonwoo committed:
Feat : 댓글 생성, 조회 API 구현

- 특정 post에 댓글을 달 수 있도록 구현하였습니다
- 특정 post에 달려있는 댓글들을 최신 순으로 10개씩 페이지네이션으로 조회가능하도록 구현하였습니다.

kimdonwoo committed:
Feat : 댓글 수정, 삭제 API 구현

- 특정 댓글을 댓글의 주인만 수정, 삭제가능하도록 구현하였습니다.
- 수정시에 기존 댓글이랑 동일한 데이터가 일때 예외처리 하였습니다.

Jun2-Lee committed:
Feat : CORS 설정을 위한 파일 추가
- CORS 문제를 해결하기 위한 Filter 추가.
- 현재는 모든 url에 대해 열어두었습니다.

kimdonwoo committed:
Merge pull request #19 from Step3-kakao-tech-campus/junhee

CORS 설정을 위한 파일 추가

kimdonwoo committed:
Merge branch '6-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into donwoo

INUK-ai committed:
Feat: 그룹 초대 링크 확인 기능

- 비공식 비공개 그룹 초대 링크 확인 기능 구현

kimdonwoo committed:
Feat : 페이지와 관련된 API에 Member랑 Group 연결

- 기존 페이지와 관련된 API 구축할 때, Member랑 Group 관련 로직을 제외하고 구현했는데 추가해주었습니다.
- 모든 API에서 요청을 보낸 유저가 해당 그룹에 속한 유저가 맞는지 확인합니다.

kimdonwoo committed:
Feat : Post(글)와 관련된 API에 Member랑 Group 연결

- 기존 글과 관련된 API 구축할 때, Member랑 Group 관련 로직을 제외하고 구현했는데 추가해주었습니다.
- 모든 글 관련 API에서 요청을 보낸 유저가 해당 그룹에 속한 유저가 맞는지 확인합니다.

kimdonwoo committed:
Feat : Comment(댓글)와 관련된 API에 Member랑 Group 연결

- 기존 댓글과 관련된 API 구축할 때, Member랑 Group 관련 로직을 제외하고 구현했는데 추가해주었습니다.
- 모든 댓글 관련 API에서 요청을 보낸 유저가 해당 그룹에 속한 유저가 맞는지 확인합니다.

kimdonwoo committed:
Feat : 전역 예외 처리 형식 통일

- 팀원간에 전역 예외 처리 방식이 상이하여 수정해서 통일해주었습니다.
- 예외 발생시 response body에 status도 추가해주었습니다.

INUK-ai committed:
Feat: 비공식 공개 그룹 추가 검색 결과

- 비공식 공개 그룹 추가 검색에 대한 결과 반환

INUK-ai committed:
Feat: 공식 그룹 추가 검색

- 공식 그룹 추가 검색 반환

INUK-ai committed:
Refactor: 내 문서 기여 목록 페이지네이션

- 그룹 내 마이 페이지, 내 문서 기여 목록 10개만 던짐
- 전체 보기를 따로

INUK-ai committed:
Refactor: 페이지네이션 수정

- 기본 검색 이후 추가 검색이라 page 기본값을 1로 수정, 그대로 받음

INUK-ai committed:
Refactor: Transactional 관리

- Transactional 빼먹은 부분 보완

kimdonwoo committed:
Merge pull request #21 from Step3-kakao-tech-campus/donwoo

6주차 기능 구현

INUK-ai committed:
Merge branch '6-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into inuk

INUK-ai committed:
Merge pull request #23 from Step3-kakao-tech-campus/inuk

6주차 기능 구현

kimdonwoo committed:
Merge pull request #24 from Step3-kakao-tech-campus/6-week

[6주차] 6-week -> develop branch 병합

INUK-ai committed:
Refactor: 그룹 설정 수정

- 각 API마다 잘못 가져온 Group 수정

INUK-ai committed:
Feat: 문서 기여 목록 페이지 구현

- 내 문서 기여 목록 전체 보기 기능 임시 구현

INUK-ai committed:
Merge branch 'develop' of https://github.com/Step3-kakao-tech-campus/Team8_BE into inuk

INUK-ai committed:
Refactor: Token에서 회원 ID 가져오기

- currentMember()를 통해 회원 ID를 가져옴

INUK-ai committed:
Feat: 에러 핸들링

- 모든 service의 API들을 try-catch문으로 감쌌음

INUK-ai committed:
Refactor: RedisConfig 수정

- <String, String>에서 <String, Object>로 수정

INUK-ai committed:
Refactor: Redis를 활용한 getMember, getGroup

- 이게 성능 개선엔 큰 도움이 될 것 같지는 않아서 우선 주석 처리

INUK-ai committed:
Refactor: 예외 처리 수정

- try-catch로 묶어서 Exception 처리한 부분 싹 다 고침

kimdonwoo committed:
Feat : 페이지 링크 거는 기능 redis로 성능 개선

- 페이지 생성시에 groupId_pageName을 Key로 하고 Value 값은 pageId를 넣어주었습니다.
- 페이지 링크 거는 기능의 성능을 위해 데이터베이스에 쿼리를 날리지 않고 redis를 통해서 페이지 링크를 가져올 수 있게 처리하였습니다.
- 페이지 삭제시에 redis에서도 삭제시켜주었습니다.

INUK-ai committed:
Fix: Invitation 연관관계 수정

- 연관관계 설정
- 초대 링크 구현 방법 바꿀 생각이라 임시로 해놨습니다.

kimdonwoo committed:
Feat : 글 신고 기능 구현 완료

- 특정 글에 대해서 신고 하는 기능 구현완료했습니다.
- 기존 Post랑 연관관계 매핑이 되어 있었는데, History로 수정했습니다.

kimdonwoo committed:
Merge pull request #28 from Step3-kakao-tech-campus/inuk

예외 처리 및 초대 링크 임시 구현

kimdonwoo committed:
Merge branch '7-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into donwoo

kimdonwoo committed:
Merge pull request #29 from Step3-kakao-tech-campus/donwoo

7주차 기능 구현

kimdonwoo committed:
Merge pull request #30 from Step3-kakao-tech-campus/7-week

[7주차] 7-week -> develop branch 병합

INUK-ai committed:
Refactor: Group 생성자 Builder에서 memberCount 기본값 1로 고정

- 생성 시 생성자만 있기에 1로 고정

INUK-ai committed:
Refactor: 초대 링크 구현 방식 변경

- Redis를 통한 초대 링크 생성, 저장 및 확인

INUK-ai committed:
Feat: 주기적으로 만료된 초대 링크 확인 후 삭제 기능 구현

- 2주에 한 번, 새벽 4시에 만료된 초대 링크 관련 데이터들을 삭제

INUK-ai committed:
Feat: 초대 링크 확인 기능 구현

- 초대 링크를 통한 접근 시 유효한 초대 링크인지 확인

kimdonwoo committed:
Feat : 엔티티 양방향 매핑 구현

- 엔티티들 사이에 단방향 매핑에서 양방향 매핑으로 수정하였습니다.

kimdonwoo committed:
Fix : HistoryJPARepository 에러 처리

- 자료구조 관련해서 에러 처리하였습니다.

kimdonwoo committed:
Style : PostResponse 변수명 수정

- history 관련해서 변수명 수정하였습니다.

Jun2-Lee committed:
Merge pull request #33 from Step3-kakao-tech-campus/donwoo

[8주차] 양방향 매핑 구현

INUK-ai committed:
Merge branch '8-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into inuk

INUK-ai committed:
Refactor: 프로젝트 패키지 구조 변경

- Group 내부 패키지 구조 변경

INUK-ai committed:
Feat: 그룹 내 그룹원 리스트 조회 기능 구현

- 양방향 매핑으로 Group 조회 후 GroupMember 리스트 반환

INUK-ai committed:
Refactor: 그룹 중복 가입 방지

- 그룹 중복 가입 예외 처리

INUK-ai committed:
Refactor: 예외 처리 추가 구현

- PostMapping의 유효성 검사 추가
- EntityNotForupException 수정

INUK-ai committed:
Refactor: ActiveGroupMember 관련 로직 수정

- Member, Group 객체 대신 MemberId와 GroupId로 Repository 접근

INUK-ai committed:
Refactor: 그룹원 조회 기능 수정

- 현재 활동중인 회원과 탈퇴한 회원 모두를 반환하는 것을 활동중인 회원만 반환하는 것으로 수정

kimdonwoo committed:
Feat : Post랑 Comment 양방향 매핑으로 변경에 따른 api 수정

- Comment 조회 부분에서 fetch join으로 comment 조회시 groupMember 도 같이 들고와서 1+N 문제해결
- Comment 생성시에 Post에 addComment 편의 메소드 추가
- Comment 삭제, 수정시 fetch join으로 groupMember도 같이 들고와 쿼리 최적화진행

kimdonwoo committed:
Feat : Post랑 History,Comment 양방향 매핑으로 변경에 따른 api 수정

- Post 생성시에 pageInfo에 편의 메서드 구현으로 List<Post>에 추가해줌
- Post 생성,수정시 History도 생성해서 List<History>에 추가해줌

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/page/PageInfoResponse.java:
```diff
@@ -59,4 +62,37 @@ public searchPageDTO(PageInfo pageInfo){
 
     }
 
+    @Getter
```
DTO도 setter 보다는 빌더 또는 생성자를 사용하는 것도 고려해보세요.

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/page/PageRestController.java:
```diff
@@ -16,14 +17,18 @@ public class PageRestController {
 
     @GetMapping("/page/{pageid}")
-    public void getPageFromId(@PathVariable Long pageid) {
+    public ResponseEntity<?> getPageFromId(@PathVariable Long pageid) {
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/history/History.java:
```diff
@@ -25,18 +24,23 @@ public class History {
     private String content;
     private LocalDateTime created_at;
 
     @Builder
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/post/PostService.java:
```diff
@@ -56,7 +63,13 @@ public PostResponse.createPostDTO createPost(Long pageId, Long parentPostId, int
         Post savedPost = postJPARepository.save(newPost);
 
-        // 7. return DTO
+        // 7. 히스토리 생성
```
👍

kimdonwoo committed:
Feat : PageInfo랑 Post의 양방향 매핑에 따른 API 수정

- 페이지 조회시 fetch join을 활용하여 성능 개선
- 페이지 생성,조회시 where문에 groupId도 추가
- 페이지 검색시 fetch join으로 1+N 문제 해결

kimdonwoo committed:
Refactor : ResponseEntitiy<?>에 타입을 정확히 명시

- 멘토님 피드백으로 ResponseEntitiy<?>에 타입을 정확히 명시해주었습니다.

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/post/PostService.java:
```diff
@@ -67,9 +65,43 @@ public PostResponse.createPostDTO createPost(Long pageId, Long parentPostId, int
     }
+
+    @Transactional
+    public PostResponse.modifyPostDTO modifyPost(Long postId , String title, String content){
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/_core/errors/ErrorCode.java:
```diff
@@ -9,7 +9,10 @@
     INTERNAL_SERVER_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "WEKIKI-000", "Internal server error"),
     PAGE_NOT_FOUND(HttpStatus.NOT_FOUND,"PAGE-001" , "존재하지 않는 페이지 입니다."),
-    PARENT_POST_NOT_FOUND(HttpStatus.NOT_FOUND,"POST-001","존재하지 않는 PARENT POST 입니다.");
+    PARENT_POST_NOT_FOUND(HttpStatus.NOT_FOUND,"POST-001","존재하지 않는 PARENT POST 입니다."),
```
적절하게 ErrorCode들을 사용하여 처리하고 있습니다 👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/page/PageService.java:
```diff
@@ -134,6 +136,29 @@ public List<PageInfoResponse.searchPageDTO> searchPage(int pageNo, String keywor
+    // ============================groupId 필요==========================================================
+
+    @Transactional
+    public PageInfoResponse.getRecentPageDTO getRecentPage(Long userId , Long groupId){
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/page/PageService.java:
```diff
@@ -61,7 +61,10 @@ public PageInfoResponse.createPageDTO createPage(String title, Long groupId, Lon
         // 2. groupMember 존재하는지 확인 (없으면 Exception)
 
-        // TODO : 3. 그룹 내 동일한 Page가 존재하는지 체크
+        // 3. 그룹 내 동일한 title의 Page가 존재하는지 체크 (TODO : where 문에 groupId 추가)
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/page/PageService.java:
```diff
@@ -28,6 +28,29 @@ public class PageService {
+        }
+
+        // 4. 포스트가 하나도 없으면 삭제시키기
+        PageInfoResponse.deletePageDTO response = new PageInfoResponse.deletePageDTO(pageInfo);
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/history/HistoryJPARepository.java:
```diff
@@ -18,4 +20,11 @@ public interface HistoryJPARepository extends JpaRepository<History, Long> {
 
     List<History> findAllByGroupMember(GroupMember groupMember);
+
+    @Query("SELECT h FROM History h JOIN FETCH h.groupMember m WHERE h.post.id = :postId ORDER BY h.created_at DESC")
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/group/GroupService.java:
```diff
@@ -128,6 +126,28 @@ public SearchGroupDTO searchGroupByKeyword(String keyword) {
+        - 확인 후 해당하는 그룹 Id 반환
+        - 가입 url 생성 후 그룹 가입 API
+     */
+    public ValidateInvitationResponseDTO ValidateInvitation(String invitationLink) {
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/page/PageJPARepository.java:
```diff
@@ -5,10 +5,12 @@
+@Repository
 public interface PageJPARepository extends JpaRepository<PageInfo, Long> {
 
     @Query("SELECT p FROM PageInfo p WHERE p.pageName LIKE :keyword%")
```
많이 언급 드렸던 부분이지만 해당 부분을 추후에 querydsl로 리팩토링해보세요 ~!

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/comment/CommentController.java:
```diff
@@ -37,6 +37,31 @@ public ResponseEntity<?> createComment(@PathVariable Long groupid,
         return ResponseEntity.ok(ApiUtils.success(response));
     }
 
+    @DeleteMapping("/comment/{commentid}")
```
👍

jinyochoi commented on wekiki/src/main/java/com/kakao/techcampus/wekiki/_core/utils/CORSFilter.java:
```diff
@@ -0,0 +1,44 @@
+
+@Component
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class CORSFilter implements Filter {
```
👍

jinyochoi reviewed (approved):
충돌나는 코드를 해결하여주세요 !

kimdonwoo committed:
Feat : 페이지 조회시에 response 추가

- 페이지 조회할 api 호출시에 response에 order랑 parentPostId 추가해줬습니다.

INUK-ai committed:
Refactor: DTO 수정

- requestDTO, responseDTO를 한 class에 모아놓음

INUK-ai committed:
Refactor: DTO 패키지 이동

- 패키지 구조 단순화

Jun2-Lee committed:
Feat : 메인 페이지, 내 정보 페이지 구현
- 메인 페이지는 JWT 토큰 확인 후 로그인 여부로 구분한 뒤 정보 전달
- 내 정보 페이지는 그대로 구현

INUK-ai committed:
Fix: DTO 디렉토리 구조 오류 수정

- 제대로 Replace가 안된 오류 수정

INUK-ai committed:
Fix: RedisUtils 생성

- Redistemplate을 바로 주입하는게 동작을 안해서 RedisUtils를 새로 Component로 생성

Jun2-Lee committed:
Merge pull request #36 from Step3-kakao-tech-campus/donwoo

[8주차] 양방향 매핑에 따른 API 수정

INUK-ai committed:
Fix: 요청 오류 수정

- DTO : @RequestBody 빠진거 추가
- Query : @Param 빠진거 추가

INUK-ai committed:
Merge branch '8-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into inuk

Jun2-Lee committed:
Merge pull request #38 from Step3-kakao-tech-campus/inuk

[8주차] 기능 구현

Jun2-Lee committed:
Fix : pull 충돌 해결

Jun2-Lee committed:
Fix : pull 충돌 해결2

kimdonwoo committed:
Merge pull request #37 from Step3-kakao-tech-campus/junhee

[8주차] 메인 페이지, 내 정보 페이지 구현

INUK-ai committed:
Fix: 초대 링크 오류 해결

- LocalDateTime 직렬화 오류 해결

INUK-ai committed:
Refactor: 모든 응답 DTO에 각 Entity의 Id 추가

- 모든 GroupResponseDTO에 GroupId 추가
- History 관련에서도 HistoryId 추가

Jun2-Lee committed:
Feat : 메인 페이지에서 내가 속하지 않은 그룹만 표시
- 로그인 후 메인페이지 접근 시, 그룹 둘러보기 창에 내가 속하지 않은 그룹만 표시되게 변경

Jun2-Lee committed:
Refactor : id를 groupId로 수정

Jun2-Lee committed:
Feat : kakao 인가 code 를 받아 kakao에서 회원 정보 조회 기능
- 아직 테스트는 마치지 않았습니다... 인가 코드를 받을 수 없어서...
- 회원가입을 어떻게 해야 할 지 고민입니다..

INUK-ai committed:
Refactor: 예외 처리와 응답 DTO 수정

- 유효성 검사 추가
- 응답 DTO 프론트와 협의 후 수정

Jun2-Lee committed:
Feat : 회원 전체 닉네임 변경 기능 추가

kimdonwoo committed:
Feat : 키워드를 통한 페이지 검색 api 수정

- 해당 키워드로 시작하는 페이지만 검색하다가 해당 키워드를 포함하는 모든 페이지 검색으로 수정하였습니다.

kimdonwoo committed:
Feat : 데이터베이스 칼럼 타입 수정

- Post의 content랑 History의 content 부분 String에서 TEXT로 수정하였습니다.

kimdonwoo committed:
deploy : 배포를 위한 설정 파일 추가

- 카카오 크램플론 배포를 위해 설정 파일 추가하였습니다.

kimdonwoo committed:
Feat : 배포환경, 테스트환경 별로 yml 설정

- 환경에 다른 yml 파일 적용되도록 설정하였습니다.

INUK-ai committed:
Refactor: 그룹 내 닉네임 중복 시 예외 처리

- 그룹 내 똑같은 닉네임 이미 있을 경우 예외 처리

INUK-ai committed:
Refactor: 그룹 생성 시 초대링크 생성 후 response에 추가

- 그룹 생성 시 초대 링크 생성 후 반환

INUK-ai committed:
Refactor: 그룹 상세 조회 시 그룹 유형 추가

- 그룹 상세 조회 시 response에 GroupType 추가

kimdonwoo committed:
Merge pull request #39 from Step3-kakao-tech-campus/donwoo

[9주차] 배포를 위한 설정 파일 수정 및 추가

INUK-ai committed:
Merge branch '8-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into inuk

INUK-ai committed:
Merge pull request #40 from Step3-kakao-tech-campus/donwoo

[9주차] 배포환경, 테스트환경 별로 yml 설정

INUK-ai committed:
Merge branch '8-week' of https://github.com/Step3-kakao-tech-campus/Team8_BE into inuk

INUK-ai committed:
Refactor: API 명세 수정 반영

- API 명세 수정 요청에 의한 response 수정

INUK-ai committed:
Refactor: 초대 링크를 통한 접근 시 responseDTO 수정

- 초대 링크를 통한 접근 성공 시 groupName도 추가 반환

Jun2-Lee committed:
Merge pull request #43 from Step3-kakao-tech-campus/inuk

[9주차] 에러 수정 및 API 명세 수정에 따른 코드 수정

Jun2-Lee committed:
Merge branch '8-week' into junhee

INUK-ai committed:
Merge pull request #44 from Step3-kakao-tech-campus/junhee

배포를 위한 Pull Request

kimdonwoo committed:
setting : yml 파일 수정

- 배포 yml이랑 로컬 yml 수정하였습니다.

kimdonwoo committed:
Merge pull request #45 from Step3-kakao-tech-campus/donwoo

setting : yml 파일 수정

kimdonwoo committed:
Merge pull request #46 from Step3-kakao-tech-campus/8-week

[9주차] 배포를 위한 8-week -> develop 병합

INUK-ai committed:
Refactor: 내 문서 기여 목록 응답DTO 1차 수정

- 인덱스 가져오는 부분 빼고 API 명세 수정에 따른 responseDTO 수정

INUK-ai committed:
Merge pull request #48 from Step3-kakao-tech-campus/inuk

Refactor: 내 문서 기여 목록 응답DTO 1차 수정
